docs/scripts/buildApi.ts

```diff
@@ -240,22 +280,438 @@ function extractEvents(eventsObject: TypeDoc.DeclarationReflection, apisToGenera
+    inheritance: reactApi.inheritance,
+    demos: generateDemoList(reactApi.demos),
+  };
+
+  // docs/pages/component-name.json
+  writePrettifiedFile(
+    path.resolve(outputDirectory, `${kebabCase(reactApi.name)}.json`),
+    JSON.stringify(pageContent),
+    prettierConfigPath,
+  );
+
+  // docs/pages/component-name.js
+  writePrettifiedFile(
+    path.resolve(outputDirectory, `${kebabCase(reactApi.name)}.js`),
+    `import * as React from 'react';
+import ApiPage from 'docsx/src/modules/components/ApiPage';
+import mapApiPageTranslations from 'docs/src/modules/utils/mapApiPageTranslations';
+import jsonPageContent from './${kebabCase(reactApi.name)}.json';
+
+export default function Page(props) {
+  const { descriptions, pageContent } = props;
+  return <ApiPage descriptions={descriptions} pageContent={pageContent} />;
+}
+
+Page.getInitialProps = () => {
+  const req = require.context(
+    'docsx/translations/api-docs/data-grid', 
+    false,
+    /${kebabCase(reactApi.name)}.*.json$/,
+  );
+  const descriptions = mapApiPageTranslations(req);
+
+  return {
+    descriptions,
+    pageContent: jsonPageContent,
+  };
+};
+  `.replace(/\r?\n/g, reactApi.EOL),
+    prettierConfigPath,
+  );
+
+  // eslint-disable-next-line no-console
+  console.log('Built API docs for', reactApi.name);
+}
+
+async function run(argv: { outputDirectory?: string }) {
   const outputDirectory = path.resolve(argv.outputDirectory!);
-  mkdirSync(outputDirectory, { mode: 0o777, recursive: true });
+  fse.mkdirSync(outputDirectory, { mode: 0o777, recursive: true });
+
+  const pagesMarkdown = findPagesMarkdown()
+    .map((markdown) => {
+      const markdownSource = fse.readFileSync(markdown.filename, 'utf8');
+      return {
+        ...markdown,
+        components: getHeaders(markdownSource).components,
+      };
+    })
+    .filter((markdown) => markdown.components.length > 0);
 
   const workspaceRoot = path.resolve(__dirname, '../../');
   const prettierConfigPath = path.join(workspaceRoot, 'prettier.config.js');
+  const tsconfig = ttp.loadConfig(path.resolve(__dirname, '../../tsconfig.json'));
 
-  const app = new TypeDoc.Application();
-  app.options.addReader(new TypeDoc.TSConfigReader());
-  app.options.addReader(new TypeDoc.TypeDocReader());
-  app.bootstrap({
-    entryPoints: ['packages/grid/data-grid/src/index.ts'],
-    exclude: ['**/*.test.ts'],
-    tsconfig: 'packages/grid/data-grid/tsconfig.json',
-  });
-  const project = app.convert();
+  const componentsToGenerateDocs = [
+    path.resolve(__dirname, '../../packages/grid/data-grid/src/DataGrid.tsx'),
+    path.resolve(__dirname, '../../packages/grid/x-grid/src/DataGridPro.tsx'),
+  ];
+
+  const indexPath = path.resolve(__dirname, '../../packages/grid/_modules_/index.ts');
+  const program = ttp.createTSProgram([...componentsToGenerateDocs, indexPath], tsconfig);
+
+  // Uncomment below to generate documentation for all exported components
```

Comment:
I prefer to only document now `DataGrid` and `DataGridPro`. There're a few small components we could merge together into bigger and meaninful components. Not documenting them allows us to change them without declaring breaking changes.